crates/ethereum-rpc/src/lib.rs

```diff
@@ -749,15 +735,21 @@ fn register_rpc_methods<C: sov_modules_api::Context, Da: DaService>(
                 match mempool_only {
                     // only ask sequencer
                     Ok(Some(true)) => {
-                        let tx = ethereum
+                        match ethereum
                             .sequencer_client
                             .as_ref()
                             .unwrap()
                             .get_tx_by_hash(hash, Some(true))
                             .await
-                            .map_err(|e| to_jsonrpsee_error_object(e, ETH_RPC_ERROR))?;
-
-                        Ok::<Option<reth_rpc_types::Transaction>, ErrorObjectOwned>(tx)
+                        {
+                            Ok(tx) => {
+                                Ok::<Option<reth_rpc_types::Transaction>, ErrorObjectOwned>(tx)
```

Comment:
I don't like the compiler can't deduce generic types for Ok/Err. Is it really the case here?